include/mbedtls/md.h

```diff
@@ -49,6 +49,12 @@ typedef enum {
     MBEDTLS_MD_RIPEMD160,
 } mbedtls_md_type_t;
 
+typedef enum {
+    MBEDTLS_SIG_RSA=0,
+    MBEDTLS_SIG_ECDSA,
+    MBEDTLS_SIG_NONE,
+} mbedtls_sig_type_t;
```

Comment:
@mpg From my understanding, elements of `mbedtls_pk_type_t` are public key types together with the information on what kind of operations (encryption, signing, key exchange) they can be used for. Signature algorithms can be extracted from them, and conversely we have the embedding `mbedtls_ssl_pk_alg_from_sig` picking for any signature algorithm one representative pubkey which supports it, but I think it might bring some clarification to introduce a designated type for signature algorithms only. For example, when the client checks whether the signature algorithm in the server's `ServerKeyExchange` matches the server's certificate, it wraps the signature algorithm into a `mbedtls_pk_type_t` and uses `pk_can_do` to check whether there's a match. This looks a bit roundabout to me, and I'd find it clearer to check whether the provided signature algorithm is the underlying signature algorithm of the server's pubkey.